Brokerages/Alpaca/AlpacaBrokerage.cs

```diff
@@ -0,0 +1,310 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using NodaTime;
+using QuantConnect.Data;
+using QuantConnect.Data.Market;
+using QuantConnect.Interfaces;
+using QuantConnect.Logging;
+using QuantConnect.Packets;
+using QuantConnect.Securities;
+using HistoryRequest = QuantConnect.Data.HistoryRequest;
+using Order = QuantConnect.Orders.Order;
+
+namespace QuantConnect.Brokerages.Alpaca
+{
+    /// <summary>
+    /// Alpaca Brokerage implementation
+    /// </summary>
+    [BrokerageFactory(typeof(AlpacaBrokerageFactory))]
+    public class AlpacaBrokerage : Brokerage, IDataQueueHandler
+    {
+        //private readonly OandaSymbolMapper _symbolMapper = new OandaSymbolMapper();
+        private readonly AlpacaApiBase _api;
```

Comment:
Why does `AlpacaApiBase` derive from `Brokerage` *and* `AlpacaBrokerage` derive from `Brokerage`? I think only `AlpacaBrokerage` should derive from `Brokerage` as the `AlpacaApiBase` isn't actually an `IBrokerage` w/ an associated factory. Perhaps what was intended was to have `AlpacaBrokerage` derive from `AlpacaApiBase` ?

Reply:
Also noting that many methods here simply pass through to `AlpacaApiBase` it looks like this type should probably derive from `AlpacaApiBase` -- thoughts?

Reply:
Well, there is no need to derive AlpacaApiBase from AlpacaBrokerage. It was done early in the implementation and I'll remove that.

Reply:
I'm suggesting that `AlpacaBrokerage : AlpacaApiBase` -- then many of the pass-through methods in `AlpacaBrokerage` can simply be deleted (along with the `_api` member variable.

Reply:
That being said, I agree that there's no reason for `AlpacaApiBase` to subclass `Brokerage` and prefer the compositional approach this PR uses (exception for the extra bit of inheritance by `AlpacaApiBase : Brokerage`)

Reply:
the new class hierarchy looks better, less simple patching of calls through, resolving.